src/request.ts

```diff
@@ -45,10 +51,13 @@ export default class Request {
     return promise;
   }
 
-  private _fetch(url: string, options: Object) : Promise<any> {
+  private _fetch(url: string, opts: FetchOptions) : Promise<any> {
     return new Promise((resolve, reject) => {
+      // Clone options since we are changing the object
+      let options : RequestInit = cloneDeep(opts as RequestInit)
```

Comment:
I'm fine if we don't want to do it that way, more of a future proofing in case it's called directly by someone passing a long-lived object.